lib/Maskings/Maskings.cpp

```diff
@@ -200,112 +200,106 @@ bool Maskings::shouldDumpData(std::string const& name) {
-      return VPackValue(data.getDouble());
+    if (data.isBool() || data.isString() || data.isInteger() ||
+        data.isDouble()) {
+      out.add(data);
+      return;
     }
   } else {
     if (data.isBool()) {
-      return func->mask(data.getBool(), buffer);
+      func->mask(data.getBool(), out, buffer);
+      return;
     } else if (data.isString()) {
-      velocypack::ValueLength length;
-      char const* c = data.getString(length);
-      return func->mask(std::string(c, length), buffer);
+      func->mask(data.stringView(), out, buffer);
+      return;
     } else if (data.isInteger()) {
-      return func->mask(data.getInt(), buffer);
+      func->mask(data.getInt(), out, buffer);
+      return;
     } else if (data.isDouble()) {
-      return func->mask(data.getDouble(), buffer);
+      func->mask(data.getDouble(), out, buffer);
+      return;
     }
   }
-  return VPackValue(VPackValueType::Null);
+
+  out.add(VPackValue(VPackValueType::Null));
 }
 
 void Maskings::addMaskedArray(Collection const& collection,
-                              VPackBuilder& builder,
-                              std::vector<std::string>& path,
-                              VPackSlice const& data) const {
-  std::string buffer;
-
-  for (VPackSlice entry : VPackArrayIterator(data)) {
-    if (entry.isObject()) {
-      VPackObjectBuilder ob(&builder);
-      addMaskedObject(collection, builder, path, entry);
-    } else if (entry.isArray()) {
-      VPackArrayBuilder ap(&builder);
-      addMaskedArray(collection, builder, path, entry);
+                              std::vector<std::string_view>& path,
+                              velocypack::Slice data, VPackBuilder& out,
+                              std::string& buffer) const {
+  for (VPackSlice value : VPackArrayIterator(data)) {
+    if (value.isObject()) {
+      VPackObjectBuilder ob(&out);
+      addMaskedObject(collection, path, value, out, buffer);
+    } else if (value.isArray()) {
+      VPackArrayBuilder ap(&out);
+      addMaskedArray(collection, path, value, out, buffer);
     } else {
-      builder.add(maskedItem(collection, path, buffer, entry));
+      maskedItem(collection, path, value, out, buffer);
     }
   }
 }
 
 void Maskings::addMaskedObject(Collection const& collection,
-                               VPackBuilder& builder,
-                               std::vector<std::string>& path,
-                               VPackSlice const& data) const {
-  std::string buffer;
-
-  for (auto const& entry : VPackObjectIterator(data, false)) {
-    std::string key = entry.key.copyString();
-    VPackSlice const& value = entry.value;
+                               std::vector<std::string_view>& path,
+                               velocypack::Slice data, VPackBuilder& out,
+                               std::string& buffer) const {
+  for (auto entry : VPackObjectIterator(data, false)) {
+    auto key = entry.key.stringView();
+    VPackSlice value = entry.value;
```

Comment:
NIT I don't sure but can we use same style
```suggestion
    velocypack::Slice value = entry.value;
```